### PR TITLE
hex: 0.20.5 -> 0.21.1

### PR DESCRIPTION
**Pull request by afontaine**
###### Motivation for this change

It looks like the bump is needed for when OTP 24 lands, as well as some new options and features

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by afontaine** [comment]
@minijackson @hauleth @NobbZ @happysalada @balsoft @yurrriq @cw789 you might be interested in this

**Comment by afontaine** [comment]
although you have to build and test it via the `elixir` package

**Review by minijackson — APPROVED** [review]
I cherry-picked your changes, and can confirm I can compile hex, fetch mix dependencies, and build an Elixir release

**Comment by happysalada** [comment]
Here is the changelog for those interested.
https://github.com/hexpm/hex/blob/main/CHANGELOG.md

**Comment by happysalada** [comment]
@balsoft I think this is ready to merge when you have a moment.

**Comment by cw789** [comment]
Hex Version **0.21.2** is already out (https://github.com/hexpm/hex/releases).

**sha256** (from `nix-prefetch-url --unpack https://github.com/hexpm/hex/archive
/v0.21.2.tar.gz`):
`18vwrc5b7pyi3nifmx5hd5wbz8fy3h6sfvkmskjg5acmz66fys0g`

Not sure if this is really the needed SHA - this always confuses me.

**Comment by happysalada** [comment]
I had completely forgotten this one. I'm going to merge this.
@cw789 would you care to do another PR?

**Comment by cw789** [comment]
@happysalada Hi Raphael. Congratulations to your merge rights. I'm really happy about it.
In this case - of course with pleasure.

**Comment by happysalada** [comment]
Thanks, and for the record, feel free to ping me if you have anything elixir related!

**Comment by afontaine** [comment]
Awesome! Thanks so much @happysalada! Congrats! If I have any more elixir PRs you'll be the first to know 🙂

And thanks @cw789!

I'll admit this PR fell off my radar for a bit, thanks to the both of you for moving this along!

**Comment by happysalada** [comment]
Yes, same here, feel free to ping me If I forget. If i haven't responded for a week, then pinging is the right approach.

**Pull request merged**
